src/request.ts

```diff
@@ -14,6 +14,11 @@ type RequestReturn = {
   headers: Headers;
 }
 
+export type EventStreamReturn = {
```

Comment:
It took me a sec to get this but I like it. How about `EventStreamHandle`?

I also read `onClose` as being a function to be called when the event stream is closed, rather than a promise resolved when it is closed (which I agree is the better interface). How about calling it just `closed`?

Reply:
Oh I like that better. Will update!